datadotworld/client.py

```diff
@@ -41,52 +47,344 @@ def __init__(self, token=None, propsfile="~/.data.world",
             raise RuntimeError((
                 'you must either provide an API token to this constructor, or create a '
                 '.data.world file in your home directory with your API token'))
+
         self.protocol = protocol
-        self.queryHost = queryHost
+        self.query_host = query_host
+        self.apiHost = api_host
+
+        self._api_client = ApiClient(host="{}://{}/v0".format(protocol, api_host), header_name='Authorization',
+                                     header_value='Bearer {}'.format(token))
+        self._datasets_api = DatasetsApi(self._api_client)
+        self._uploads_api = UploadsApi(self._api_client)
+
+    # Dataset Operations
+
+    def get_dataset(self, dataset_key=None):
```

Comment:
Well... I thought defining keyword arguments would aid usability/improve clarity, but you are right. `None` isn't a good default. I'll fix that.